api/v1alpha1/instancegroup_types.go

```diff
@@ -111,6 +111,12 @@ var (
 	LifecycleHookAllowedDefaultResult   = []string{LifecycleHookResultAbandon, LifecycleHookResultContinue}
 	LaunchTemplatePlacementTenancyTypes = []string{HostPlacementTenancyType, DefaultPlacementTenancyType, DedicatedPlacementTenancyType}
 	log                                 = ctrl.Log.WithName("v1alpha1")
+
+	volumeTypesWithProvisionedIopsSupport = map[string]bool{
```

Comment:
minor nit:
Should we move this to awsprovider along with `AllowedVolumeTypes`?
We might want a shorter name too - how about `AllowedVolumeTypesWithProvisionedIOPS`?

Also what are you thoughts on using a string slice (such as `AllowedVolumeTypes`) vs. a map like you implemented here?

We can then remove the map check and change the if condition to something like:
```yaml
if v.Iops != 0 && !common.ContainsEqualFold(awsprovider.AllowedVolumeTypesWithProvisionedIOPS, v.Type) {
  ...
}
```

This is minor and both options are good, but just might be more in line with existing code.

Reply:
Sounds good - will make this check!